packages/journey-manager/src/index.ts

```diff
@@ -493,7 +525,27 @@ export const run = async (props: RunProps): Promise<RunOutput> => {
    * Change detection
    */
 
-  const documentObserver = new MutationObserver(() => {
+  const documentObserver = new MutationObserver((mutations) => {
+    // If any of the added nodes are inside matches on appear events, trigger those events
+    mutations.forEach(async (mutation) => {
+      const targets = await withElements(appearSteps);
```

Comment:
```suggestion
    const targets = await withElements(appearSteps);
    mutations.forEach(async (mutation) => {
```

Seems like a performance win, no?